### PR TITLE
refactor: replace explicit spans with `#[instrument]` in peer_actor

### PR DESCRIPTION
**Pull request by nagisa**
cc #10663
cc @robin-near

### CODE REVIEW

File: chain/network/src/peer/peer_actor.rs [hunk 1]
```diff
@@ -392,13 +392,14 @@ impl PeerActor {
         self.send_message_with_encoding(msg, Encoding::Borsh);
     }
 
+    #[tracing::instrument(
+        level = "trace",
+        target = "network",
+        "send_message_with_encoding",
+        skip_all,
+        fields(msg_type = msg.msg_variant())
+    )]
```
Comment (wacban) [inline chain/network/src/peer/peer_actor.rs:401]: Can you give me a quick TLDR of what it does?

**Review by wacban — COMMENTED** [review]
LGTM but the CI is rather unhappy. Can you have a look?

File: chain/network/src/peer/peer_actor.rs (hunk 1 above)
Comment (nagisa) [reply chain/network/src/peer/peer_actor.rs:401]: This is pretty much a different way to write `tracing::span!` that covers the entire function and happens correctly handle `async fn` by default. I updated the style guide with some information about it today: https://near.github.io/nearcore/practices/style.html#spans

**Comment by robin-near** [comment]
Thanks @nagisa for doing what I didn't find pleasant to do. The instrument macro's syntax is puzzling (so is tracing::span).

**Comment by nagisa** [comment]
> Attention: Patch coverage is 57.94393% with 45 lines in your changes are missing coverage. Please review.

Thanks bot, this is an NFC that happens to change the indentation of many lines.

**Review by wacban — APPROVED** [review]

**Pull request merged**